### PR TITLE
openresty: 1.21.4.3 -> 1.25.3.1

### PR DESCRIPTION
**Pull request by lblasc**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by afh** [comment]
Result of `nixpkgs-review pr 280223` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>openresty</li>
    <li>openresty.doc</li>
  </ul>
</details>

File: pkgs/servers/http/openresty/default.nix [hunk 1]
```diff
@@ -40,7 +40,8 @@ callPackage ../nginx/generic.nix args rec {
   configureFlags = [ "--with-http_postgres_module" ];
 
   postInstall = ''
-    ln -s $out/luajit/bin/luajit-2.1.0-beta3 $out/bin/luajit-openresty
+    ln -s $out/luajit/bin/luajit-2.1.ROLLING $out/bin/luajit-openresty
+    ln -sf $out/nginx/bin/nginx $out/bin/openresty
```
Comment (afh) [inline pkgs/servers/http/openresty/default.nix:44]: Can you provide more context about addition of this line symlinking nginx to openresty, @lblasc?

Comment (lblasc) [reply pkgs/servers/http/openresty/default.nix:44]: If I recall correctly, build will produce broken  `$out/bin/openresty` symlink, this line fixes that.

Comment (kalekseev) [reply pkgs/servers/http/openresty/default.nix:44]: Tried to build it and I confirm that the link is broken and this symlink update fixes it.
Original link points at bin/nginx:
```
❯ ls -la /nix/store/y1qy6lhscfcihbwkxfh9w3farsbigiyb-openresty-1.25.3.1/bin/
total 336
dr-xr-xr-x  11 root  wheel    352 Jan  1  1970 .
dr-xr-xr-x  12 root  wheel    384 Jan  1  1970 ..
lrwxr-xr-x   1 root  wheel     32 Jan  1  1970 luajit-openresty -> ../luajit/bin/luajit-2.1.ROLLING
-r-xr-xr-x   1 root  wheel  19232 Jan  1  1970 md2pod.pl
lrwxr-xr-x   1 root  wheel     18 Jan  1  1970 nginx -> ../nginx/bin/nginx
-r-xr-xr-x   1 root  wheel  16041 Jan  1  1970 nginx-xml2pod
lrwxr-xr-x   1 root  wheel      9 Jan  1  1970 openresty -> bin/nginx
-r-xr-xr-x   1 root  wheel  63697 Jan  1  1970 opm
-r-xr-xr-x   1 root  wheel  36900 Jan  1  1970 resty
-r-xr-xr-x   1 root  wheel  15004 Jan  1  1970 restydoc
-r-xr-xr-x   1 root  wheel   8920 Jan  1  1970 restydoc-index
```

Comment (afh) [reply pkgs/servers/http/openresty/default.nix:44]: Thanks for the context, @lblasc, and thanks for chiming in and investigating, @kalekseev, much appreciated!

**Comment by lblasc** [comment]
I think it's time to hit the merge button, @SuperSandro2000 can you maybe do it ?

**Pull request merged**